File: ruby/move_zeroes.rb
# frozen_string_literal: true

# @param {Integer[]} nums
# @return {Void} Do not return anything, modify nums in-place instead.
def move_zeroes(nums)
  i = 0
  j = 0

  while j < nums.length
    if nums[j] != 0
      nums[i], nums[j] = nums[j], nums[i]
      i += 1
    end

    j += 1
  end

  while i < nums.length
    nums[i] = 0
    i += 1
  end

  nums
end

File: typescript/moveZeroes.ts
function moveZeroes(nums: number[]): void {
  let count = 0;

  for (let i = 0; i < nums.length; i++) {
    if (nums[i] !== 0) {
      let temp = nums[count];
      nums[count] = nums[i];
      nums[i] = temp;
      count++;
    }
  }
}
